crates/cfg-noodle/src/lib.rs

```diff
@@ -266,6 +266,11 @@ pub trait NdlDataStorage {
     /// being written.
     async fn push(&mut self, data: &Elem<'_>) -> Result<usize, Self::Error>;
 
+
+
+    /// Read raw data from storage into `buf` at `offset`.
+    async fn read_raw_data(&mut self, _offset: usize, _buf: &mut [u8]) -> Result<(), ()> { Err(())}
```

Comment:
This crate is crafted with carefully considered APIs. Let's continue that.
```suggestion
    /// Read raw data from storage into `buf` at `offset`. This API is there to provide the ability to do dumps/backups
    async fn read_raw_data(&mut self, offset: usize, buf: &mut [u8]) -> Result<(), Self::Error>;
```